### PR TITLE
ENG-1609: Add tagging_instructions field to DataCategory for LLM classification

### PR DESCRIPTION
**Pull request by thabofletcher**
Contributes to [ENG-1609]

### Description Of Changes

This PR adds support for storing LLM classification instructions directly on records in the Fides database. This enables per-category customization of how the LLM classifier should tag data, supporting both custom taxonomies and overrides of default fideslang categories.

The  field stores text that will be injected into LLM prompts to guide classification behavior for each data category. This is consumed by fidesplus's LLM classifier to provide dynamic, DB-driven taxonomy instructions.

### Code Changes

**Database & Model:**
* Added `tagging_instructions` TEXT column to `ctl_data_categories` table via Alembic migration
* Updated `DataCategory` SQLAlchemy model to include the new column
* Migration includes both upgrade and downgrade paths

**Schemas:**
* Added `tagging_instructions` field to `DataCategory` response schema
* Added `tagging_instructions` field to `DataCategoryCreateOrUpdate` request schema
* Field is optional (nullable) to support gradual adoption

**API Endpoints:**
* **PATCH** `/api/v1/data_category/{fides_key}/tagging_instructions` - Update/add tagging instructions for a specific category
* **DELETE** `/api/v1/data_category/{fides_key}/tagging_instructions` - Remove tagging instructions (set to null)
* **PUT** `/api/v1/data_category` - Existing endpoint now accepts `tagging_instructions` in payload
* **GET** `/api/v1/data_category` - Existing endpoint now returns `tagging_instructions` in response

**Tests:**
* Comprehensive test suite covering all CRUD operations
* Tests for authentication and authorization (scopes)
* Tests for edge cases (non-existent categories, null values, overwrites)

### Steps to Confirm

1. Run the migration: `alembic upgrade head`
2. Verify the `tagging_instructions` column exists in `ctl_data_categories`:
   ```sql
   SELECT column_name, data_type FROM information_schema.columns 
   WHERE table_name='ctl_data_categories' AND column_name='tagging_instructions';
   ```
3. Test the PATCH endpoint:
   ```bash
   curl -X PATCH http://localhost:8080/api/v1/data_category/user.contact.email/tagging_instructions \
     -H "Authorization: Bearer $TOKEN" \
     -H "Content-Type: application/json" \
     -d '{"tagging_instructions": "* DO TAG: email addresses\n* DO NOT TAG: encrypted emails"}'
   ```
4. Test the DELETE endpoint:
   ```bash
   curl -X DELETE http://localhost:8080/api/v1/data_category/user.contact.email/tagging_instructions \
     -H "Authorization: Bearer $TOKEN"
   ```
5. Verify GET includes the field:
   ```bash
   curl http://localhost:8080/api/v1/data_category?fides_key=user.contact.email \
     -H "Authorization: Bearer $TOKEN"
   ```

### Pre-Merge Checklist

* [x] Issue requirements met
* [ ] All CI pipelines succeeded
* [x] `CHANGELOG.md` updated
  * [x] Add a db-migration label to the entry
  * [ ] Add a high-risk label - no high-risk changes
  * [ ] Updates unreleased work already in Changelog, no new entry necessary
* Followup issues:
  * [ ] Followup issues created
  * [x] No followup issues
* Database migrations:
  * [x] Ensure that your downrev is up to date with the latest revision on `main`
  * [x] Ensure that your `downgrade()` migration is correct and works
    * Downgrade tested and removes the column successfully
  * [ ] No migrations
* Documentation:
  * [ ] Documentation complete, PR opened in fidesdocs
  * [x] Documentation issue created in fidesdocs
  * [ ] If there are any new client scopes created - **No new scopes created**
  * [ ] No documentation updates required

[ENG-1609]: https://ethyca.atlassian.net/browse/ENG-1609?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/fides/api/schemas/taxonomy_extensions.py [hunk 1]
```diff
@@ -44,6 +48,10 @@ class DataUseCreateOrUpdate(TaxonomyCreateOrUpdateBase):
 
 class DataCategoryCreateOrUpdate(TaxonomyCreateOrUpdateBase):
     parent_key: Optional[FidesKey] = None
+    tagging_instructions: Optional[str] = Field(
+        default=None,
+        description="Instructions for LLM-based data classification tagging.",
+    )
```
Comment (adamsachs) [inline src/fides/api/schemas/taxonomy_extensions.py:54]: nice - so by adding this, we support updating the field via the existing `PUT /api/v1/data_category` endpoint, as well as the new endpoints you added. i think that makes sense 👍

File: src/fides/api/api/v1/endpoints/generic_overrides.py [hunk 2]
```diff
@@ -516,6 +516,75 @@ async def update_data_category(
...
 
 
+@data_category_router.patch(
+    "/data_category/{fides_key}/tagging_instructions",
+    dependencies=[Security(verify_oauth_client, scopes=[DATA_CATEGORY_UPDATE])],
+    response_model=DataCategory,
+    status_code=status.HTTP_200_OK,
+    name="Update tagging instructions",
+)
+async def update_data_category_tagging_instructions(
```
Comment (adamsachs) [inline src/fides/api/api/v1/endpoints/generic_overrides.py:526]: ok realizing now this follows the pattern of other endpoints in this module, but the use of `async` here with an underlying service that uses a non-`async` db session is a footgun. using an `async` fastAPI endpoint with underlying non-`async` code that is potentially long running can effectively lock up the entire webserver.

your operations here should be very quick, so there's little risk in this particular update - it's just highlighting what i think is a broad risk in this module...i'll raise that up with the wider team separately!

**Review by adamsachs — APPROVED** [review]
these updates look solid to me 👍
